src/com/goide/actions/GoCreateFileAction.java

```diff
@@ -41,7 +44,12 @@
   protected PsiFile createFile(String name, @NotNull String templateName, @NotNull PsiDirectory dir) {
     FileTemplate template = FileTemplateManager.getInstance(dir.getProject()).getInternalTemplate(templateName);
     Properties properties = new Properties();
-    properties.setProperty(PACKAGE, ContainerUtil.getLastItem(StringUtil.split(dir.getName(), "-")));
+    Collection<String> packages = GoUtil.getAllPackagesInDirectory(dir);
```

Comment:
It makes sense just to getFirstItem and check for null instead of checking isEmpty and then hope that nothing will be changed in future and getFirstItem will return notnull value.